Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -348,31 +347,8 @@ public IAsyncLoop Mine(WalletSecret walletSecret)
                     if (!this.connection.ConnectedNodes.Any()) this.logger.LogTrace("Waiting to be connected with at least one network peer...");
                     else this.logger.LogTrace("Waiting for IBD to complete...");
 
-                    tryToSync = true;
                     await Task.Delay(TimeSpan.FromMilliseconds(this.minerSleep), this.nodeLifetime.ApplicationStopping).ConfigureAwait(false);
-                }
-
-                // TODO: What is the purpose of this conditional block?
-                // It seems that if it ends with continue, the next round of the loop will just 
-                // not execute the above while loop and it won't even enter this block again,
-                // so it seems like no operation.
-                // In StratisX there is the wait uncommented. Then it makes sense. Why we have it commented?
-                if (tryToSync)
-                {
-                    tryToSync = false;
-                    // TODO: This condition to have at least 3 peers is disqualifying us on testnet quite often.
-                    // Yet the 60 secs delay does not prevent us to mine because tryToSync will be false next time we are here
-                    // unless we are completely disconnected. So this is weird logic.
-                    bool fewPeers = this.connection.ConnectedNodes.Count() < 3;
-                    bool lastBlockTooOld = chainTip.Header.Time < (this.dateTimeProvider.GetTime() - 10 * 60);
-                    if ((fewPeers && !this.network.IsTest()) || lastBlockTooOld)
-                    {
-                        if (fewPeers) this.logger.LogTrace("Node is connected to few peers.");
-                        if (lastBlockTooOld) this.logger.LogTrace("Last block is too old, timestamp {0}.", chainTip.Header.Time);
-
-                        await Task.Delay(TimeSpan.FromMilliseconds(60000), this.nodeLifetime.ApplicationStopping).ConfigureAwait(false);
-                        continue;
-                    }
+                    continue;
```

Comment:
What's the purpose of this continue here? Don't you want to loop until we're connected and not in IBD?

Reply:
Sorry, this is wrong. Will fix that.

Reply:
So, now it should be better - the continue obviously has no purpose there, but what I intended to do is to continue with the external loop, not internal. I've fixed that by reordering the code. So now when we exit the loop (after we waited for IBD to complete) we check if consensus is aligned with the tip.